woocommerce/woocommerce-ios
# [Woo POS][Cash & Receipts] Render cash button and navigate to cash view

Closes: #14603
Closes: #14599

## Description
This PR adds the `cash payment` button to the POS dashboard, and the initial navigation back and forth between both screens.

Please disregard the UI details on the cash view, we'll address the design when implementing the cash functionality and error handling.

## Changes
- Adds two new payment states: `.acceptingCash` and `.cashPaymentSuccessful`
- Adds a `PointOfSaleCollectCashView`
- As with Android, we present the view as standalone view, rendered from the POS dashboard. Upon tapping in cash collection we switch the payment state, which drives rendering the new view, upon navigating back, we return to the previous payment state which restores the dashboard as it was.

## Testing
- Enable `.acceptCashForPointOfSale` 
- Run the POS on a physical device
- Start to process a cart checkout, with a disconnected reader, observe the `cash payment` buttons appears and it's tappable below the reader not connected error:

![Simulator Screenshot - iPad Air 11 - iOS 17 5 M2 - 2024-12-19 at 17 19 10](https://github.com/user-attachments/assets/15b0a6f6-2351-4090-88d8-e0fffdc85c8b)

- Observe the order total appears on the the right top corner below the navigation button

<img width="232" alt="Screenshot 2024-12-19 at 17 20 06" src="https://github.com/user-attachments/assets/5eec57ee-6375-4203-9201-0f652b4e6395" />


- Connect the reader and observe the button still renders. Tapping the button and navigating back and forth retains the payment state, so the reader should remain connected:

https://github.com/user-attachments/assets/50b25ced-a2a8-4651-ad13-0c168a82c4ae

- At the moment "Mark payment as complete" is not functional.

## Screenshots


---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
